OPEN_PROBLEMS/ENHANCED_BITSWAP_GRAPHSYNC.md

```diff
@@ -0,0 +1,156 @@
+
+## Solving this Open Problem
+
+### What is the impact
+
+Bitswap is a central component of the IPFS system and a protocol that generally influences the performance of IPFS as a whole. Designing and integrating smart algorithmic solutions to improve its performance can be instrumental for the performance and adoption of IPFS. It is worth highlighting that bitswap allows ample space for improvement, hence, several different solutions are encouraged and will be considered. Addressing sub-graphs of the DAG through IPLD as GraphSync is attempting to do in a secure manner is a first necessary step.
+
+### What defines a complete solution?
+> What hard constraints should it obey? Are there additional soft constraints that a solution would ideally obey?
+
+First and foremost, any complete solution should account for extensibility as the IPFS system needs to scale up and more applications are implemented on top. The active number of users of IPFS is increasing exponentially and the requests submitted to the network are following accordingly. That being said and acknowledging the fact that it is often difficult to find "one size fits all" solutions, tunable versions of bitswap that can dynamically adapt according to network and environment conditions is an avenue that needs to be explored.
+
+There are several desirable features discussed within IPFS that should be implemented in a complete solution for bitswap and its successor GraphSync - see list below in Existing Conversations/Threads section. Some of them target clean protocol extensions, while some others target higher-level measurement and UI/UX issues.
+
+## Other
+
+### Existing Conversations/Threads
+
+- [Batch gets](https://github.com/ipfs/notes/issues/285)
+- [Move giant files/datasets at Level1 + Level2 ](https://github.com/ipfs/notes/issues/218)
+- [Reed-Solomon layer over IPFS](https://github.com/ipfs/notes/issues/196)
+- [Tracking progress and other data](https://github.com/ipfs/notes/issues/107)
+- [Advanced Pinning](https://github.com/ipfs/notes/issues/49)
+- [bandwidth estimators](https://github.com/ipfs/notes/issues/30)
+- [Alternative BitSwap strategies](https://github.com/ipfs/notes/issues/20)
+- [2-Hop Bitswap](https://github.com/ipfs/notes/issues/386)
+- [IPFS Spy](https://github.com/jimpick/go-ipfs/blob/jim/network-logging/README.ipfs-spy.md)
+- [Bitswap Simulator](https://github.com/heems/bssim)
+- [bsdash - dashboard for go-ipfs bitswap](https://www.npmjs.com/package/bsdash)
+
+- Presentation of bitswap in Sept 2019 IPFS Weekly Call: https://www.youtube.com/watch?v=G_Q7iTpwYQU&list=PLuhRWgmPaHtSGRSHdU9dbsukHKlihZZAe&index=4
+
+
+### Extra notes
+
+[qri.io](https://qri.io/): a data transfer mechanism using IPFS components to keep track of blocks in a DAG using Manifest files (similar to bittorrent magnet files) - https://github.com/qri-io/dag
```

Comment:
This should go as "one of the experiments within the IPFS Ecosystem", it is a tool that uses IPFS and its APIs for faster syncs.